sdk/keyvault/Azure.Security.KeyVault.Administration/src/KeyVaultSetting.cs

```diff
@@ -21,59 +28,37 @@ public KeyVaultSetting(string name, bool value)
             Argument.AssertNotNullOrEmpty(name, nameof(name));
 
             Name = name;
-            Type = SettingType.Boolean;
-
-            // bool.ToString() returns "True" or "False", but the service wants "true" or "false".
-            Value = value ? "true" : "false";
+            Value = new KeyVaultSettingValue(value);
         }
 
-        // TODO: Consider using Azure.Value and making this class mutable: https://github.com/Azure/azure-sdk-for-net/issues/32174
-
         /// <summary>
-        /// Gets the boolean value of this account setting if <see cref="Type"/> is <see cref="SettingType.Boolean"/>.
+        /// Creates a new instance of the <see cref="KeyVaultSetting"/> class.
         /// </summary>
-        /// <returns>A boolean value if <see cref="Type"/> is <see cref="SettingType.Boolean"/>.</returns>
-        /// <exception cref="InvalidOperationException">The <see cref="Type"/> is not <see cref="SettingType.Boolean"/>, or the value cannot be normalized as a Boolean.</exception>
-        public bool AsBoolean() => CheckType(SettingType.Boolean, () => (bool.TryParse(Value, out bool parsedValue), parsedValue));
+        /// <param name="name">The name of the account setting.</param>
+        /// <param name="value">The boolean value of the account setting.</param>
+        /// <param name="settingType">The type specifier of the value.</param>
+        internal KeyVaultSetting(string name, string value, KeyVaultSettingType? settingType)
```

Comment:
The comment calls it a `boolean value` but its typed as `string`?

Reply:
Oops